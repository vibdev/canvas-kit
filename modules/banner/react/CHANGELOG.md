# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.25](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.24...@workday/canvas-kit-react-banner@0.1.25) (2019-06-20)


### Bug Fixes

* broad stroke accessibility fixes ([9803412](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/commits/9803412))





## [0.1.24](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.23...@workday/canvas-kit-react-banner@0.1.24) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.23](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.22...@workday/canvas-kit-react-banner@0.1.23) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.22](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.21...@workday/canvas-kit-react-banner@0.1.22) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.21](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.20...@workday/canvas-kit-react-banner@0.1.21) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.20](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.19...@workday/canvas-kit-react-banner@0.1.20) (2019-06-14)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.19](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.18...@workday/canvas-kit-react-banner@0.1.19) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.18](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.17...@workday/canvas-kit-react-banner@0.1.18) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.17](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.16...@workday/canvas-kit-react-banner@0.1.17) (2019-06-10)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.16](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.15...@workday/canvas-kit-react-banner@0.1.16) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.15](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.14...@workday/canvas-kit-react-banner@0.1.15) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.14](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.13...@workday/canvas-kit-react-banner@0.1.14) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.13](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.12...@workday/canvas-kit-react-banner@0.1.13) (2019-05-31)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.12](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.11...@workday/canvas-kit-react-banner@0.1.12) (2019-05-24)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.11](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.10...@workday/canvas-kit-react-banner@0.1.11) (2019-05-22)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.10](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.9...@workday/canvas-kit-react-banner@0.1.10) (2019-05-21)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.9](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.8...@workday/canvas-kit-react-banner@0.1.9) (2019-05-15)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.7...@workday/canvas-kit-react-banner@0.1.8) (2019-05-14)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.7](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.6...@workday/canvas-kit-react-banner@0.1.7) (2019-05-10)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.5...@workday/canvas-kit-react-banner@0.1.6) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.5](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.4...@workday/canvas-kit-react-banner@0.1.5) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.4](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.3...@workday/canvas-kit-react-banner@0.1.4) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.2...@workday/canvas-kit-react-banner@0.1.3) (2019-05-07)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.1...@workday/canvas-kit-react-banner@0.1.2) (2019-05-06)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## 0.1.1 (2019-04-30)

**Note:** Version bump only for package @workday/canvas-kit-react-banner
