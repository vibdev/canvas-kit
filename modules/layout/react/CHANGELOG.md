# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.17](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.16...@workday/canvas-kit-react-layout@0.1.17) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.16](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.15...@workday/canvas-kit-react-layout@0.1.16) (2019-06-14)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.15](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.14...@workday/canvas-kit-react-layout@0.1.15) (2019-06-10)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.14](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.13...@workday/canvas-kit-react-layout@0.1.14) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.13](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.12...@workday/canvas-kit-react-layout@0.1.13) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.12](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.11...@workday/canvas-kit-react-layout@0.1.12) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.11](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.10...@workday/canvas-kit-react-layout@0.1.11) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.10](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.9...@workday/canvas-kit-react-layout@0.1.10) (2019-05-31)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.9](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.8...@workday/canvas-kit-react-layout@0.1.9) (2019-05-24)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.7...@workday/canvas-kit-react-layout@0.1.8) (2019-05-22)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.7](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.6...@workday/canvas-kit-react-layout@0.1.7) (2019-05-21)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.5...@workday/canvas-kit-react-layout@0.1.6) (2019-05-15)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.5](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.4...@workday/canvas-kit-react-layout@0.1.5) (2019-05-14)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.4](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.3...@workday/canvas-kit-react-layout@0.1.4) (2019-05-10)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.2...@workday/canvas-kit-react-layout@0.1.3) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.1...@workday/canvas-kit-react-layout@0.1.2) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.1](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.0...@workday/canvas-kit-react-layout@0.1.1) (2019-05-07)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





# 0.1.0 (2019-04-10)


### Features

* **layout:** added fixed width prop for columns ([fe4a860](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/fe4a860))
* **layouts:** added canvas spacing to layouts ([607ed59](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/607ed59))
* **layouts:** added spacing feature ([f388d0d](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/f388d0d))
* **layouts:** content -> capWidth ([93efa87](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/93efa87))
* **layouts:** fixed type errors ([463447a](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/463447a))
* **layouts:** initial commit for layouts component ([ae29eae](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/ae29eae))
* **layouts:** limited columns to 1 through 12 ([3cbacfb](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/3cbacfb))
* **layouts:** made column children optional ([26831f3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/26831f3))
* **layouts:** removed padding from first and last element on column ([9a3e9d3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/9a3e9d3))
* **layouts:** updated readme ([e10218d](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/e10218d))
* **layouts:** updated readme ([8fdc4ce](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/8fdc4ce))
