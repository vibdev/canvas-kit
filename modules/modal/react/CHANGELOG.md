# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/modal/compare/@workday/canvas-kit-react-modal@0.0.1...@workday/canvas-kit-react-modal@0.0.2) (2019-06-20)

**Note:** Version bump only for package @workday/canvas-kit-react-modal





## 0.0.1 (2019-06-20)

**Note:** Version bump only for package @workday/canvas-kit-react-modal
