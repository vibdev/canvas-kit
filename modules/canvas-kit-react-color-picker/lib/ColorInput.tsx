import * as React from 'react';
import styled from 'react-emotion';
import {focusRing} from '@workday/canvas-kit-react-common';
import {colors, spacing, type, InputProvider} from '@workday/canvas-kit-react-core';
import {css} from 'emotion';
import {checkSmallIcon} from '@workday/canvas-system-icons-web';
import {SystemIcon} from '@workday/canvas-kit-react-icon';
import {pickDarkOrLightColor} from './ColorUtils';

export interface ColorInputState {
  isInputFocused: boolean;
  typedInHexValue: string;
}

export interface ColorInputProps {
  onEnterPress: (color: string) => void;
  selectedHexColor: string;
  showSwatchTileCheckIcon?: boolean;
}

const swatchTileSpacing = spacing.xxs;
const swatchTileSize = 20;
const swatchCheckIconSpacing = 8;

const CustomHexInput = styled('input')({
  margin: spacing.zero,
  height: spacing.xl,
  WebkitAppearance: 'none',
  MozAppearance: 'none',
  borderColor: 'transparent',
  borderRadius: '4px',
  border: `1px solid ${colors.frenchVanilla400}`,
  boxSizing: 'border-box',
  textAlign: 'left',
  paddingLeft: '46px',
  marginRight: spacing.xxs,
  [`[data-whatinput='mouse'] &:focus,
  [data-whatinput='keyboard'] &:focus,
  [data-whatinput='pointer'] &:focus`]: {
    outline: 'none',
    borderColor: 'transparent',
    ...focusRing(2, 0),
  },
});

const ColorInputContainer = styled('div')({
  display: 'flex',
  position: 'relative',
});

const PoundSignPrefix = styled('span')({
  position: 'absolute',
  left: 36,
  top: 12,
  ...type.small,
  color: colors.blackPepper100,
});

const SwatchTile = styled('div')({
  position: 'absolute',
  cursor: 'pointer',
  height: swatchTileSize,
  width: swatchTileSize,
  top: spacing.zero,
  bottom: spacing.zero,
  left: swatchTileSpacing,
  margin: 'auto',
  boxShadow: 'inset 0 0 0 1px rgba(0,0,0,0.25)',
  pointerEvents: 'none',
  borderRadius: '2px',
});

const swatchCheckIcon = css({
  position: 'absolute',
  left: 6,
  top: swatchCheckIconSpacing,
});

export default class ColorInput extends React.Component<ColorInputProps, ColorInputState> {
  public constructor(props: ColorInputProps) {
    super(props);
    this.state = {
      typedInHexValue: this.props.selectedHexColor,
      isInputFocused: false,
    };
  }

  public render() {
    const {selectedHexColor, showSwatchTileCheckIcon} = this.props;
    const {typedInHexValue, isInputFocused} = this.state;
    return (
      <InputProvider>
        <ColorInputContainer>
          <PoundSignPrefix>#</PoundSignPrefix>
          <CustomHexInput
            onKeyPress={this.onKeyPress}
            onChange={this.onChange}
            type="text"
            placeholder="FFFFFF"
            defaultValue={typedInHexValue}
          />
          <SwatchTile style={{backgroundColor: `${typedInHexValue || ''}`}} />
          {selectedHexColor && isInputFocused && showSwatchTileCheckIcon ? (
            <SystemIcon
              fill={pickDarkOrLightColor(selectedHexColor)}
              fillHover={pickDarkOrLightColor(selectedHexColor)}
              className={swatchCheckIcon}
              icon={checkSmallIcon}
            />
          ) : null}
        </ColorInputContainer>
      </InputProvider>
    );
  }

  private onKeyPress = (evt: React.KeyboardEvent<HTMLInputElement>) => {
    const isValidHex = this.isValidHexValue(this.state.typedInHexValue);
    if (evt.key === 'Enter' && isValidHex) {
      this.handleSubmit();
    }
  };

  private onChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
    const isValidHex = this.isValidHexValue(evt.target.value);
    const newColorHexValue = isValidHex ? this.addPoundSign(evt.target.value) : '';
    this.setState({
      typedInHexValue: newColorHexValue,
      isInputFocused: false,
    });
  };

  private isValidHexValue = (hexCode: string): boolean => {
    return /(^#?[0-9A-F]{6}$)|(^#?[0-9A-F]{3}$)/i.test(hexCode);
  };

  private addPoundSign = (hexCode: string): string => {
    return `#${hexCode}`;
  };

  private handleSubmit = (): void => {
    this.setState({
      isInputFocused: true,
    });
    this.props.onEnterPress(this.state.typedInHexValue);
  };
}
