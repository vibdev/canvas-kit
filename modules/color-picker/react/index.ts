import ColorInput from './lib/ColorInput';
import ColorPreview from './lib/ColorPreview';

export default ColorInput;
export {ColorInput, ColorPreview};
export * from './lib/ColorInput';
export * from './lib/ColorPreview';
