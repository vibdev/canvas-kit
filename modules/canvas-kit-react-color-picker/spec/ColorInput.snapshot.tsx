import * as React from 'react';
import ColorInput from '../lib/ColorInput';
import * as renderer from 'react-test-renderer';

describe('ColorInput Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(
      <ColorInput onChange={jest.fn()} onValidColorChange={jest.fn()} value={''} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders with a selected color', () => {
    const component = renderer.create(
      <ColorInput onChange={jest.fn()} onValidColorChange={jest.fn()} value={'e6e'} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders with a selected color without #', () => {
    const component = renderer.create(
      <ColorInput onChange={jest.fn()} onValidColorChange={jest.fn()} value={'#e6e'} />
    );
    expect(component).toMatchSnapshot();
  });
});
