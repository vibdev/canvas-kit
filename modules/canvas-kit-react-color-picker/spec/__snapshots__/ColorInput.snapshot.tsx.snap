// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorInput Snapshots renders a dark checkIcon if hex value is light 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.emotion-0 {
  position: absolute;
  left: 36px;
  top: 10px;
  font-size: 14px;
  color: #787878;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-2 {
  margin: 0px;
  height: 40px;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-color: transparent;
  border-radius: 4px;
  border: 1px solid #a6a6a6;
  box-sizing: border-box;
  padding-left: 46px;
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-right: 8px;
}

[data-whatinput='mouse'] .emotion-2:focus,
[data-whatinput='keyboard'] .emotion-2:focus,
[data-whatinput='pointer'] .emotion-2:focus {
  outline: none;
  border-color: transparent;
  -webkit-animation: animation-bvpr5k 100ms;
  animation: animation-bvpr5k 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #005cb9;
}

.emotion-4 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0px;
  bottom: 0px;
  left: 8px;
  margin: auto;
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.25);
  pointer-events: none;
  border-radius: 2px;
}

<div
  data-whatinput="initial"
  data-whatintent={null}
>
  <div
    className="emotion-6 emotion-7"
  >
    <span
      className="emotion-0 emotion-1"
    >
      #
    </span>
    <input
      className="emotion-2 emotion-3"
      onChange={[Function]}
      placeholder="FFFFFF"
      type="text"
      value="ffffff"
    />
    <div
      className="emotion-4 emotion-5"
      style={
        Object {
          "backgroundColor": "",
        }
      }
    />
  </div>
</div>
`;

exports[`ColorInput Snapshots renders a light checkIcon if hex value is dark 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.emotion-0 {
  position: absolute;
  left: 36px;
  top: 10px;
  font-size: 14px;
  color: #787878;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-2 {
  margin: 0px;
  height: 40px;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-color: transparent;
  border-radius: 4px;
  border: 1px solid #a6a6a6;
  box-sizing: border-box;
  padding-left: 46px;
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-right: 8px;
}

[data-whatinput='mouse'] .emotion-2:focus,
[data-whatinput='keyboard'] .emotion-2:focus,
[data-whatinput='pointer'] .emotion-2:focus {
  outline: none;
  border-color: transparent;
  -webkit-animation: animation-bvpr5k 100ms;
  animation: animation-bvpr5k 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #005cb9;
}

.emotion-4 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0px;
  bottom: 0px;
  left: 8px;
  margin: auto;
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.25);
  pointer-events: none;
  border-radius: 2px;
}

<div
  data-whatinput="initial"
  data-whatintent={null}
>
  <div
    className="emotion-6 emotion-7"
  >
    <span
      className="emotion-0 emotion-1"
    >
      #
    </span>
    <input
      className="emotion-2 emotion-3"
      onChange={[Function]}
      placeholder="FFFFFF"
      type="text"
      value="e6e"
    />
    <div
      className="emotion-4 emotion-5"
      style={
        Object {
          "backgroundColor": "",
        }
      }
    />
  </div>
</div>
`;

exports[`ColorInput Snapshots renders as expected 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.emotion-0 {
  position: absolute;
  left: 36px;
  top: 10px;
  font-size: 14px;
  color: #787878;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-2 {
  margin: 0px;
  height: 40px;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-color: transparent;
  border-radius: 4px;
  border: 1px solid #a6a6a6;
  box-sizing: border-box;
  padding-left: 46px;
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-right: 8px;
}

[data-whatinput='mouse'] .emotion-2:focus,
[data-whatinput='keyboard'] .emotion-2:focus,
[data-whatinput='pointer'] .emotion-2:focus {
  outline: none;
  border-color: transparent;
  -webkit-animation: animation-bvpr5k 100ms;
  animation: animation-bvpr5k 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #005cb9;
}

.emotion-4 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0px;
  bottom: 0px;
  left: 8px;
  margin: auto;
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.25);
  pointer-events: none;
  border-radius: 2px;
}

<div
  data-whatinput="initial"
  data-whatintent={null}
>
  <div
    className="emotion-6 emotion-7"
  >
    <span
      className="emotion-0 emotion-1"
    >
      #
    </span>
    <input
      className="emotion-2 emotion-3"
      onChange={[Function]}
      placeholder="FFFFFF"
      type="text"
      value=""
    />
    <div
      className="emotion-4 emotion-5"
      style={
        Object {
          "backgroundColor": "",
        }
      }
    />
  </div>
</div>
`;

exports[`ColorInput Snapshots renders with a selected color 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.emotion-0 {
  position: absolute;
  left: 36px;
  top: 10px;
  font-size: 14px;
  color: #787878;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-2 {
  margin: 0px;
  height: 40px;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-color: transparent;
  border-radius: 4px;
  border: 1px solid #a6a6a6;
  box-sizing: border-box;
  padding-left: 46px;
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-right: 8px;
}

[data-whatinput='mouse'] .emotion-2:focus,
[data-whatinput='keyboard'] .emotion-2:focus,
[data-whatinput='pointer'] .emotion-2:focus {
  outline: none;
  border-color: transparent;
  -webkit-animation: animation-bvpr5k 100ms;
  animation: animation-bvpr5k 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #005cb9;
}

.emotion-4 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0px;
  bottom: 0px;
  left: 8px;
  margin: auto;
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.25);
  pointer-events: none;
  border-radius: 2px;
}

<div
  data-whatinput="initial"
  data-whatintent={null}
>
  <div
    className="emotion-6 emotion-7"
  >
    <span
      className="emotion-0 emotion-1"
    >
      #
    </span>
    <input
      className="emotion-2 emotion-3"
      onChange={[Function]}
      placeholder="FFFFFF"
      type="text"
      value="e6e"
    />
    <div
      className="emotion-4 emotion-5"
      style={
        Object {
          "backgroundColor": "",
        }
      }
    />
  </div>
</div>
`;

exports[`ColorInput Snapshots renders with a selected color without # 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.emotion-0 {
  position: absolute;
  left: 36px;
  top: 10px;
  font-size: 14px;
  color: #787878;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-2 {
  margin: 0px;
  height: 40px;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-color: transparent;
  border-radius: 4px;
  border: 1px solid #a6a6a6;
  box-sizing: border-box;
  padding-left: 46px;
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-right: 8px;
}

[data-whatinput='mouse'] .emotion-2:focus,
[data-whatinput='keyboard'] .emotion-2:focus,
[data-whatinput='pointer'] .emotion-2:focus {
  outline: none;
  border-color: transparent;
  -webkit-animation: animation-bvpr5k 100ms;
  animation: animation-bvpr5k 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #005cb9;
}

.emotion-4 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0px;
  bottom: 0px;
  left: 8px;
  margin: auto;
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.25);
  pointer-events: none;
  border-radius: 2px;
}

<div
  data-whatinput="initial"
  data-whatintent={null}
>
  <div
    className="emotion-6 emotion-7"
  >
    <span
      className="emotion-0 emotion-1"
    >
      #
    </span>
    <input
      className="emotion-2 emotion-3"
      onChange={[Function]}
      placeholder="FFFFFF"
      type="text"
      value="e6e"
    />
    <div
      className="emotion-4 emotion-5"
      style={
        Object {
          "backgroundColor": "",
        }
      }
    />
  </div>
</div>
`;
